src/sources/soundsourceffmpeg.cpp

```diff
@@ -119,98 +228,81 @@ SoundSource::OpenResult SoundSourceFFmpeg::tryOpen(const AudioSourceConfig& /*au
     const QByteArray qBAFilename(getLocalFileName().toLocal8Bit());
 #endif
 
-    // Open file and make m_pFormatCtx
-    if (avformat_open_input(&m_pFormatCtx, qBAFilename.constData(), nullptr,
-                            &l_iFormatOpts) != 0) {
-        qDebug() << "SoundSourceFFmpeg::tryOpen: cannot open" << localFileName;
+    // Open file and initialize m_pFormatCtx
+    const int avformat_open_input_result =
+            avformat_open_input(
+                    &m_pFormatCtx, qBAFilename.constData(), nullptr, nullptr);
+    if (avformat_open_input_result != 0) {
+        qWarning() << "[SoundSourceFFmpeg]"
+                << "avformat_open_input() failed and returned"
+                << avformat_open_input_result;
         return OpenResult::FAILED;
     }
 
-    // TODO() why is this required, should't it be a runtime check
-#if LIBAVCODEC_VERSION_INT > 3544932 // 54.23.100
-    av_dict_free(&l_iFormatOpts);
```

Comment:
It's written in the API docs. FFmpeg returns a dictionary with the 
unrecognized options for this in/out parameter. The "out" dictionary must be 
a subset of the "in" dictionary. Since we provide a nullptr we will never 
get anything back that needs to be freed.

On 08.09.2016 20:58, Daniel Schürmann wrote:
>
> In src/sources/soundsourceffmpeg.cpp 
> <https://github.com/mixxxdj/mixxx/pull/1003#discussion_r78068244>:
>
> >          return OpenResult::FAILED;
> >      }
> >
> > -    // TODO() why is this required, should't it be a runtime check
> > -#if LIBAVCODEC_VERSION_INT > 3544932 // 54.23.100
> > -    av_dict_free(&l_iFormatOpts);
>
> Why can this be removed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/1003/files/14399cf62b5d5ee96a4d0d3a361af8ab165b2144#r78068244>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-aoPh0i3_y5yoVcxbBzPJ1FSx4mfAks5qoFq8gaJpZM4J2QNu>.
>